eclipse-theia/theia
# feat(scm): add visibility variable in plugin API

#### What it does

Adds support to specify whether a SourceControlInputBox is visible

Contributed on behalf of STMicroelectronics

Signed-off-by: Olaf Lessenich <olessenich@eclipsesource.com>

Fixes #11140

#### How to test
1. Install the test extension [vscode-extension-sourcecontrol-inputbox-11140-0.0.1.zip](https://github.com/eclipse-theia/theia/files/9091122/vscode-extension-sourcecontrol-inputbox-11140-0.0.1.zip) in your theia workspace from your local filesystem.
2. Use the command `SourceControlInputBox: Status` which is provided by the extension to display the properties of `sourceControl.inputBox`. Verify that a Boolean `visible` property is present.
3. (Optional: Use the provided commands `SourceControlInputBox: Hide` and `SourceControlInputBox: Show` to manipulate the respective Boolean. Verify the change with `SourceControlInputBox: Status`.)
4. This extension can be installed in VSCode as well to compare the behaviour.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
